lib/farscape/agent.rb

```diff
@@ -7,11 +7,12 @@ class Agent
 
     attr_reader :media_type
     attr_reader :entry_point
-
-    def initialize(entry = nil, media = :hale, safe = false)
+    
+    def initialize(entry = nil, media = :hale, safe = false, pluginhash = {})
       @entry_point = entry
       @media_type = media
       @safe_mode = safe
+      @pluginhash = pluginhash.empty? ? default_plugin_hash : pluginhash
```

Comment:
I would properly snake case the variable name for enhanced readability: `@plugin_hash`